sqlserver/tests/test_connection.py

```diff
@@ -387,10 +423,12 @@ def test_connection_error_reporting(
     expected_error_pattern = matching_patterns[0]
 
     check = SQLServer(CHECK_NAME, {}, [instance_docker])
-    connection = Connection(check.init_config, check.instance, check.handle_service_check)
+    connection = Connection(check, check.init_config, check.instance, check.handle_service_check)
     with pytest.raises(SQLConnectionError) as excinfo:
         with connection.open_managed_default_connection():
             pytest.fail("connection should not have succeeded")
 
     message = str(excinfo.value)
     assert re.search(expected_error_pattern, message)
+    assert len(check._warnings_by_code) == 1
+    assert check._warnings_by_code[expected_warning_value] is not None
```

Comment:
I imagine that it would be tedious to check for the actual warning message strings but I think it might still be worth having _some test_ that validates that the formatting done when recording warning is done correctly. It's not super useful on a continuous basis but I almost always found something a little off in the formatted string when I've been adding new warnings.

Reply:
yeah that makes sense to me 👍 

I'll add some additional test cases for that